### ISSUE TITLE
Dependencies missing in PHAR

### ISSUE THREAD

**Issue by ogmueller** [phingofficial/phing#1053]
When phing is installed as PHAR (e.g. by phive), it will not include "suggested dependencies" as mentioned on https://packagist.org/packages/phing/phing.

So for example, if I'd like to use the StopwatchTask, it won't work, because the dependency to symfony/stopwatch is not included in the PHAR. As far as I know, there is no option to add this dependency later on.

Would it be possible to add as many "suggested dependencies" as possible to the PHAR?

**Comment by siad007** [comment]
Hi which version do you use? AFAIK there was a ticket where this was fixed  https://github.com/phingofficial/phing/issues/566
Would you mind testing it against 3 alpha 2?

**Comment by mrook** [comment]
@ogmueller that's correct. At this point the suggested packages have a purely informational value (this is a limitation of composer). I'll think about how to add those packages to the generated .phar.

**Comment by ogmueller** [comment]
FYI: I tried to install it via phive and it downloaded phing-3.0.0-alpha2.phar.

**Comment by mrook** [comment]
Ref https://github.com/phingofficial/phing/commit/76b94b1cf3cee3791ca9c4622e01e3cb7c271543

**Comment by jawira** [comment]
What about using Composer to install dependencies ? 

```
$ composer require symfony/stopwatch
```

Then import autoload file from `build.xml` using `AutoloaderTask`:

```xml
<?xml version="1.0" encoding="UTF-8" ?>

<project name="Dependencies missing in PHAR" default="stopWatch">

  <autoloader autoloaderpath="vendor/autoload.php"/>

  <target name="stopWatch">
    <stopwatch name="dummy" action="start"/>
    <sleep seconds="2"/>
    <stopwatch name="dummy" action="stop"/>
  </target>

</project>
```

This work well with **Phing 2.16.1** :

```sh
$ phing stopWatch # using Phing 2.16.1
Buildfile: /home/jawira/deleteme/build.xml
[autoloader] Loading autoloader from vendor/autoload.php

Dependencies missing in PHAR > stopWatch:

[stopwatch] Starttime: 0 - Endtime: 2002 - Duration: 2002 - Memory: 12582912
[stopwatch] Category:   
[stopwatch] Origin:     1561662053544.1
[stopwatch] Start time: 0
[stopwatch] End time:   2002
[stopwatch] Duration:   2002
[stopwatch] Memory:     12582912

BUILD FINISHED

Total time: 2.0968 seconds
```

But doesn't with **Phing 3.0.0-alpha2** :disappointed: :

```sh
$ phing stopWatch # using Phing 3.0.0-alpha2
Buildfile: /home/jawira/deleteme/build.xml
[autoloader] Loading autoloader from vendor/autoload.php

Dependencies missing in PHAR > stopWatch:


BUILD FAILED
/home/jawira/deleteme/build.xml:8:43 Trying to invoke private method StopwatchTask::start() from scope ReflectionMethod

Total time: 0.1337 seconds
```

Of course this completely misses the point of using a Phar file, but I think it's a nice workaround.

**Comment by siad007** [comment]
@jawira this looks like a bug - would you mind to open a new ticket. Also it would be very helpful to see the full stacktrace.

**Comment by jawira** [comment]
> @jawira this looks like a bug - would you mind to open a new ticket. Also it would be very helpful to see the full stacktrace.

Done #1080 :)
